src/block/types.ts

```diff
@@ -403,11 +404,17 @@ export type StyledBlockProps = {
   $whiteSpace?: Responsive<WhiteSpace>;
 };
 
+type BaseProps<P extends {}> = P & {
+  $style?: StyleObject | ((props: P) => StyleObject);
+  className?: string;
+};
+
+type AddStyletronRef<P extends { ref: any }> = P extends { ref: infer R } ? P & { ref?: R } : P;
```

Comment:
should this be `P & { $ref?: R }` like it is in styletron?

currently this helper type, does not really do anything(I somehow missed that in previous version of `BlockComponentType` as well)

Reply:
Yeah I can change to $ref. Is that what you mean by it does not do anything? Or are you saying it should be removed?

Reply:
yes, changing it to $ref will align it with styletron

(in a current state `P & { ref?: R }` is effectively the same as just `P`, because it is used only when `ref` is already in `P`)